src/core_plugins/kibana/server/routes/api/values/register_values.js

```diff
@@ -0,0 +1,57 @@
+import handleESError from '../../../lib/handle_es_error';
+
+export function registerValues(server) {
+  server.route({
+    path: '/api/kibana/values/{index}',
+    method: ['POST'],
+    handler: async function (req, reply) {
+      const { index } = req.params;
+      const { field } = req.payload;
+      const { query } = req.payload;
+
+      const { callWithRequest } = server.plugins.elasticsearch.getCluster('data');
+      const body = getBody(field, query);
+      let fieldList = [];
+      try {
+
+        const response = await callWithRequest(req, 'search', { index, body });
+        const buckets = response.aggregations.aggResults.buckets;
+        for (let i = 0; i < buckets.length; i < i++) {
+          const currentField = buckets[i].key;
+          if (typeof (currentField) === 'object') {
+            // its an array
+            fieldList = fieldList.concat(currentField);
+          } else {
+            fieldList.push(currentField);
+          }
+        }
+        const uniqueValues = Array.from(new Set(fieldList));
+        reply(uniqueValues);
+      } catch (error) {
+        reply(handleESError(error));
+      }
+    }
+  });
+}
+
+function getBody(field, { query }) {
+  const aggregationSize = 10000;
```

Comment:
> Also now it is hardcoded to a 10k limit for the value, I would like to make this configurable maybe with an advanced setting or with an argument in the filter pop up.

Yeah, I think it would be idea to support this as a parameter and default to something smaller, like 100.

Reply:
I also think the the name `Top Values` would be more appropriate for this filter, since it's always going to be limited by the size (even if we set it to something huge like 10k).

Reply:
>Yeah, I think it would be idea to support this as a parameter and default to something smaller, like 100.

What about having the default value in an advanced setting? It is easier to have a custom default value.